### PR TITLE
Make all ai turrets have the silicon faction

### LINKED ISSUE #14097: AI turrets have the wrong faction
So I fixed the bug where turrets wouldn't respect simple animal factions.

But the default for turrets is "neutral" meaning AI core turrets will basically ignore half the simple animals in the game (notably stands, and anything xenobio makes).

I could change the turret default to something like "silicon" but then any turrets people build in hallways will target Ian forever.

I don't really want to edit 7 maps AI turrets to silicon either.

What is the best solution to this?

### PR DESCRIPTION
**Pull request by optimumtact**
The neutral faction has too many things in it, rendering AI turrets bad

Fixes #14097 

edit: so I don't forget these

```
grep -r -l 'porta_turret{ai = 1}' | xargs sed -i 's/porta_turret{ai = 1}/porta_turret{ai = 1; faction = "silicon"}/'
grep -r -l 'porta_turret{ai = 1;' | xargs sed -i 's/porta_turret{ai = 1;/porta_turret{ai = 1; faction = "silicon";/'
```

### CODE REVIEW

**Comment by MrStonedOne** [comment]
It seems like this is something that should be done as a type.

**Comment by Jalleo** [comment]
Well syndidrones even better at killing AI now i think.

**Comment by optimumtact** [comment]
why do they have the silicon faction?

**Comment by KorPhaeron** [comment]
Syndrones are ignored by turrets anyway because turrets ignore drones

**Comment by Cheridan** [comment]
Yeah I agree with MSO, doing this as a map edit is asking for problems down the line when a mapper forgets to set it.

**Comment by optimumtact** [comment]
F I N E I'LL MAKE A SUBTYPE

**Comment by optimumtact** [comment]
grep -r -l 'porta_turret{ai = 1' | xargs sed -i  's#porta_turret{ai = 1; faction = "silicon"; \(dir = [0-9]\)}#porta_turret/ai{\1}#'

grep -r -l 'porta_turret{ai = 1' | xargs sed -i  's#porta_turret{ai = 1; faction = "silicon"}#porta_turret/ai#'

**Comment by optimumtact** [comment]
I created a subtype and also cleaned up the target handling for porta_turrets somewhat, they'll handle factions better now IMO.

drones will also be targetted by AI turrets, I can change this if you would like (add drones to the silicon faction)

DNM as I want to test targetting

**Comment by Cheridan** [comment]
nice uncompiling code citrus

**Comment by Cheridan** [comment]
can't wake up

**Comment by tkdrg** [comment]
@optimumtact is this ready now

**Comment by optimumtact** [comment]
still want to test the actual targetting logic

**Comment by optimumtact** [comment]
Cest la ve 

It's ready

**Comment by tkdrg** [comment]
Compile error

**Comment by optimumtact** [comment]
Whoops

**Comment by optimumtact** [comment]
There

File: code/game/machinery/portable_turret.dm [hunk 1]
```diff
@@ -467,47 +464,51 @@
...
-		if(emagged && C.stat != DEAD)	//if emagged, every living carbon is a target.
-			targets += C
-			continue
+	var/list/targets = calculate_targets()
 
-		if(C.stat || C.handcuffed || (C.lying && !(C.buckled)))	//if the perp is handcuffed or lying or dead/dying, no need to bother really
-			continue
+	if(!tryToShootAt(targets))
+		if(!always_up)
+			spawn()
```
Comment (tkdrg) [inline code/game/machinery/portable_turret.dm]: Do you really need to readd this `spawn()`? PR looks fine to me otherwise.

**Pull request merged**